### PR TITLE
UserInfo consolidation of GroupMemberInfo and MemberInfo panels

### PR DESCRIPTION
**Pull request by t3chguy**
Signed-off-by: Michael Telatynski <7t3chguy@gmail.com>

---

Fixes https://github.com/vector-im/riot-web/issues/5349
Fixes https://github.com/vector-im/riot-web/issues/5570

### CODE REVIEW

**Comment by t3chguy** [comment]
image

**Comment by t3chguy** [comment]
> Also, the feature flagged approach here has the risk that any changes to MemberInfo & co. won't be applied here. Should we rather replace MemberInfo & co. in this PR straight away?

This was discussed and is simply due to the surface area of the topic, and it proving difficult to test in our current strategies, this way we can opt-in test it to gain some confidence in it.

> It would have been nice to unify the components in the traditional class-style to have minimal changes, and in a separate commit (or PR), change to functional style, but I know all too well how easy it is to end up with big commits :)

Agreed, but the state keeping around this component was utterly awful, and basically duplicating it into the RoomAdminToolsSection component would have just been painful

File: src/components/views/right_panel/UserInfo.js [hunk 1]
```diff
@@ -0,0 +1,1212 @@
...
+            const ignoredUsers = cli.getIgnoredUsers();
+            if (isIgnored) {
+                const index = ignoredUsers.indexOf(member.userId);
+                if (index !== -1) ignoredUsers.splice(index, 1);
+            } else {
+                ignoredUsers.push(member.userId);
+            }
+
+            cli.setIgnoredUsers(ignoredUsers).then(() => {
+                // return this.setState({isIgnoring: !this.state.isIgnoring});
```
Comment (bwindels) [inline src/components/views/right_panel/UserInfo.js]: need to call `setIsIgnored` from here still I guess?

Comment (t3chguy) [reply src/components/views/right_panel/UserInfo.js]: To work around lack of local echo I guess so, the value actually comes down https://github.com/matrix-org/matrix-react-sdk/pull/3465/files#diff-a5813c8342fa7717bb0bcca63f59eacaR770

Comment (bwindels) [reply src/components/views/right_panel/UserInfo.js]: aha, that's fine then. Can we remove the commented code?

Comment (bwindels) [reply src/components/views/right_panel/UserInfo.js]: Can we still remove the commented code though?

**Pull request merged**